dev/ci/ci-common.sh

```diff
@@ -105,16 +104,52 @@ git_download()
   local giturl="${!giturl_var}"
   local ref_var="${project}_CI_REF"
   local ref="${!ref_var}"
+  local parent_project_var="${project}_CI_PARENT_PROJECT"
+  local parent_project="${!parent_project_var}"
+  local submodule_folder_var="${project}_CI_SUBMODULE_FOLDER"
+  local submodule_folder="${!submodule_folder_var}"
+  local parent_project_dest="$CI_BUILD_DIR/$parent_project"
 
   local ov_url=${overlays[${project}_URL]}
   local ov_ref=${overlays[${project}_REF]}
 
   if [ -d "$dest" ]; then
     echo "Warning: download and unpacking of $project skipped because $dest already exists."
-  elif [[ $ov_url ]] || [ "$WITH_SUBMODULES" = "1" ] || [ "$CI" = "" ]; then
-    git clone "$giturl" "$dest"
-    pushd "$dest"
-    git checkout "$ref"
+  elif [[ $ov_url ]] || [ "$WITH_SUBMODULES" = "1" ] || [ "$CI" = "" ] || [ -n "${parent_project}" ]; then
+    if [ -n "${parent_project}" ]; then
+      # if there is a parent project, we first download the parent
+      # project then symlink the submodule_folder to dest; this allows
+      # project CI scripts to be transparent w.r.t. whether or not the
+      # project is cloned from a submodule / submodule_folder.
+      if [ -d "${parent_project_dest}" ] && [ ! -e "${parent_project_dest}/.git" ]; then
+        # if the parent project does not have its .git directory (for
+        # example, when it comes from a downloaded CI artifact and the
+        # git_download above is a no-op), we must re-download it.  we
```

Comment:
why?

Reply:
Roughly the issue is that downloading the parent does not appropriately overlay unused submodules.  Let me fix that.

Reply:
I've remove the logic around .git